src/staking/delegation.service.ts

```diff
@@ -0,0 +1,109 @@
+import { HttpService } from '@nestjs/axios';
+import {
+  ForbiddenException,
+  Injectable,
+  InternalServerErrorException,
+  Logger,
+} from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import { BigNumber } from 'ethers';
+import { catchError, firstValueFrom, map } from 'rxjs';
+
+@Injectable()
+export class DelegationService {
+  private readonly logger = new Logger(DelegationService.name);
+  constructor(
+    private configService: ConfigService,
+    private readonly httpService: HttpService,
+  ) {}
+
+  async totalDelegated() {
+    const delegationApi = `${this.configService.get<string>(
+      'delegationAPIRoot',
+    )}/validators?show_fully_delegated=true&show_my_validators=false`;
+    let delegationData;
+    try {
+      const realTimeDelegationData = await firstValueFrom(
```

Comment:
I would change this a bit similar to what I did in the other project. Add headers and use a single pipe. Example:
`const headersRequest = {
  'Content-Type': 'application/json',
  'Accept-Encoding': 'gzip,deflate,compress',
};

 const realTimeDelegationData = await firstValueFrom(
        this.httpService
          .get(delegationApi, { headers: headersRequest })
          .pipe(
            map((res: AxiosResponse) => {
              return res.data;
            }),
            catchError((error) => {
              this.logger.error(`Error calling Delegation API: ${err}`);
              throw new ForbiddenException('API not available');
            }),
          )
      );`